identity/app/actions/AuthenticatedActions.scala

```diff
@@ -42,8 +40,30 @@ class AuthenticatedActions(
   def sendUserToReauthenticate(request: RequestHeader): Result =
     redirectWithReturn(request, "/reauthenticate")
 
-  def authAction: AuthenticatedBuilder[AuthenticatedUser] =
-    new AuthenticatedBuilder(authService.authenticatedUserFor, anyContentParser, sendUserToSignin)
+  def sendUserToRegister(request: RequestHeader) : Result =
+    redirectWithReturn(request, "/register")
+
+  def authRefiner: ActionRefiner[Request, AuthRequest] = new ActionRefiner[Request, AuthRequest] {
+    override val executionContext = ec
+
+    def refine[A](request: Request[A]) =
+      authService.authenticatedUserFor(request) match {
+        case Some(authenticatedUser) => Future.successful(Right(new AuthenticatedRequest(authenticatedUser, request)))
+        case None =>
+          // If an email query param exists we want to check the DB for the user and redirect them to signin/register
+          request.getQueryString("email") match {
+            case None => Future.successful(Left(sendUserToSignin(request)))
+            case Some(email) =>
+              identityApiClient.userFromQueryParam(email, "emailAddress").map {
```

Comment:
Nesting is getting a bit much here, seperate this block?  (Assuming it can't be refactored into a for-comprehension)